net: openthread: upmerge to `6edb06e`

edmont opened the pull request:
Regular OpenThread upmerge to `6edb06e`.

Also add `OPENTHREAD_CSL_RECEIVER_LOCAL_TIME_SYNC` config.

rlubos commented on west.yml:
```diff
@@ -301,7 +301,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread
-      revision: d62167ee34b091e7025c9ec2820aae71e17a3944
+      revision: pull/121/head
```
`6edb06e4e0472411200ce2a084a783eaf3faffe3`

rlubos commented on modules/openthread/Kconfig.features:
```diff
@@ -92,6 +92,14 @@ config OPENTHREAD_CSL_RECEIVER
 	help
 	  Enable CSL Receiver support for Thread 1.2
 
+
```
nit: Extra empty line

edmont committed:
net: openthread: upmerge to `6edb06e`

Regular OpenThread upmerge to `6edb06e`.

Also add `OPENTHREAD_CSL_RECEIVER_LOCAL_TIME_SYNC` config.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
